setup.py

```diff
@@ -48,213 +48,255 @@
-    author_email="contact@scalyr.com",
-    # Choose your license
-    license="Apache",
-    # See https://pypi.python.org/pypi?%3Aaction=list_classifiers
-    classifiers=[
-        # How mature is this project? Common values are
-        #   3 - Alpha
-        #   4 - Beta
-        #   5 - Production/Stable
-        "Development Status :: 5 - Production/Stable",
-        # Indicate who your project is intended for
-        "Intended Audience :: System Administrators",
-        "Topic :: System :: Logging",
-        "Topic :: System :: Monitoring",
-        # Pick your license as you wish (should match "license" above)
-        "License :: OSI Approved :: Apache Software License",
-        # Specify the Python versions you support here. In particular, ensure
-        # that you indicate whether you support Python 2, Python 3 or both.
-        "Programming Language :: Python :: 2",
-        "Programming Language :: Python :: 2.4",
-        "Programming Language :: Python :: 2.5",
-        "Programming Language :: Python :: 2.6",
-        "Programming Language :: Python :: 2.7",
-    ],
-    # What does your project relate to?
-    keywords="monitoring tools",
-    # You can just specify the packages manually here if your project is
-    # simple. Or you can use find_packages().
-    packages=find_packages(exclude=["tests*"]),
-    # List run-time dependencies here.  These will be installed by pip when your
-    # project is installed. For an analysis of "install_requires" vs pip's
-    # requirements files see:
-    # https://packaging.python.org/en/latest/technical.html#install-requires-vs-requirements-files
-    # install_requires=['peppercorn'],
-    # If there are data files included in your packages that need to be
-    # installed, specify them here.  If using Python 2.6 or less, then these
-    # have to be included in MANIFEST.in as well.
-    # package_data={
-    #    'sample': ['package_data.dat'],
-    # },
-    package_data=my_package_data,
-    # Although 'package_data' is the preferred approach, in some case you may
-    # need to place data files outside of your packages.
-    # see http://docs.python.org/3.4/distutils/setupscript.html#installing-additional-files
-    # In this case, 'data_file' will be installed into '<sys.prefix>/my_data'
-    data_files=my_data_files,
-    # To provide executable scripts, use entry points in preference to the
-    # "scripts" keyword. Entry points provide cross-platform support and allow
-    # pip to create the appropriate form of executable for the target platform.
-    # entry_points={
-    #    'console_scripts': [
-    #        'sample=sample:main',
-    #    ],
-    # },
-    console=[
-        {
-            "script": path.join("source_root", "scalyr_agent", "agent_main.py"),
-            "dest_base": "scalyr-agent-2",
-        },
-        {
-            "script": path.join("source_root", "scalyr_agent", "config_main.py"),
-            "dest_base": "scalyr-agent-2-config",
+    setup(
```

Comment:
Is ``setup()`` function now intentionally nested under ``perform_setup()`` function?

If so, I wonder if this will work under all scenarios since ``setup()`` should be directly importable.

Reply:
Yeah, I wasn't quite sure if this was the right way to do this or not.  As you can see from the rest of the PR, I need to be able to import the `WINDOWS_MONITOR_MODULES_TO_INCLUDE` and `WINDOWS_PY2_SIX_MOVES_IMPORTS` in the checker.  However, I was worried about the side effect of loading the setup module with how it was currently written.  For example, we make a copy of VERSION.  And, I didn't quite understand the impact of invoking that `setup` method.

I took this approach because, to the best of my knowledge, we always use `setup.py` as a script -- `python setup.py XXX`.  So, by triggering off of `__name__ == "main"` I thought I would preserve the current behavior.

But, if this isn't the right way to do this, I'm more than happy to change it has well.  You understand this better than I do.

Reply:
One option would also be to move those constants to another file and then reading it in setup.py and in the other Python file where they are used (we just need to be careful that the constants file doesn't import any other things from scalyr_agent since that won't be available in setup.py before install phase).

For now, I think we can leave this change as-is and in the future, if we encounter any issues, we can change it.

And as far running it as script goes - yeah that's indeed true for us, but some (packaging) tools rely on importing it (which is less of an issue these days with wheels, but still).

Reply:
Ok, thanks for that comment.  I'll go ahead and submit this as-is and we'll change it if we need it.  I suspect we will be taking a hard look at `setup.py` when we move to freezing.